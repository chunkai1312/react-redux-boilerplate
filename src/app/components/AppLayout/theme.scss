.layout {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  overflow-y: auto;

  .navDrawer {

    .drawerContent {
      border-right: 0;
      background-color: $color-white
    }

  }

  .panel {
    background-color: $palette-grey-200
  }

}