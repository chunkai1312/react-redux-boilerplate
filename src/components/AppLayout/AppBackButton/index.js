import AppBackButton from './AppBackButton'

export default AppBackButton
