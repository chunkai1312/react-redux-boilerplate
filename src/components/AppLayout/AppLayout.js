import React, { PropTypes } from 'react'
import { Link } from 'react-router'
import Helmet from 'react-helmet'
import config from '../../config'
import logo from './logo.svg'
import style from './style.css'

const AppLayout = ({ children }) => {
  return (
    <div className={style.appLayout}>
      <Helmet {...config.app.head} />
      <div className={style.appHeader}>
        <img src={logo} className={style.appLogo} alt="logo" />
        <h2>Welcome to React</h2>
      </div>
      <p className="App-intro">
        <Link to="/">Home</Link>
        {' '}
        <Link to="/about">About</Link>
      </p>
      {children}
    </div>
  )
}

AppLayout.propTypes = {
  children: PropTypes.node
}

export default AppLayout
