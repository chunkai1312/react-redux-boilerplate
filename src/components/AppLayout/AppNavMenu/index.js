import AppNavMenu from './AppNavMenu'

export default AppNavMenu
