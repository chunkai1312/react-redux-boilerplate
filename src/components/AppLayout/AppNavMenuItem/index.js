import AppNavMenuItem from './AppNavMenuItem'

export default AppNavMenuItem
