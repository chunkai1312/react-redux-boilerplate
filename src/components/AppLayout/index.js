export AppLayout from './AppLayout'
export AppTitleBar from './AppTitleBar'
export NavMenu from './NavMenu'
export NavMenuItem from './NavMenuItem'
export NavButton from './NavButton'
export BackButton from './BackButton'
export MainContent from './MainContent'

export default from './AppLayout'
