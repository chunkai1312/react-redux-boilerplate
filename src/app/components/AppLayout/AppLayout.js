import React from 'react'
import { Layout } from 'react-toolbox/lib/layout'
import theme from './theme.scss'

const AppLayout = (props) => <Layout {...props} theme={theme} />

export default AppLayout
