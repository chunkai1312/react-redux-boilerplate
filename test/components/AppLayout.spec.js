import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Helmet from 'react-helmet'
import { Link } from 'react-router'
import AppLayout from '../../src/components/AppLayout'

describe('<AboutPage />', () => {
  function setup () {
    const wrapper = shallow(<AppLayout />)
    return { wrapper }
  }

  it('should render a <Helmet /> component', () => {
    const { wrapper } = setup()
    expect(wrapper.find(Helmet)).toHaveLength(1)
  })

  it('should render two <Link /> components', () => {
    const { wrapper } = setup()
    expect(wrapper.find(Link)).toHaveLength(2)
  })

  it('should render correctly', () => {
    const { wrapper } = setup()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
