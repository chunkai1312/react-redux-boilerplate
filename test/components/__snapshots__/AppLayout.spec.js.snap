// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> should render correctly 1`] = `
<div
  className="appLayout"
>
  <HelmetWrapper
    meta={
      Array [
        Object {
          "content": "A starter kit for building universal React/Redux apps",
          "name": "description",
        },
        Object {
          "charset": "utf-8",
        },
      ]
    }
    titleTemplate="%s | Universal React Redux Starter Kit"
  />
  <div
    className="appHeader"
  >
    <img
      alt="logo"
      className="appLogo"
      src="test-file-stub"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/"
    >
      Home
    </Link>
     
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/about"
    >
      About
    </Link>
  </p>
</div>
`;
